WordPress/gutenberg
# `BoxControl`: update design

## What?


While working on adding updated sizing in #56185, we noticed that the larger sizing affects the visibility of the input for `BoxControl` when the width is constrained. And while there is a max-width on this component, the main concern is this happens regardless due to the width of the inspector:

<img width="264" alt="Screenshot 2023-11-15 at 6 32 19 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/1f766e83-bed0-4809-aa94-d85820b64303">

## Why?


`BoxControl` is only used in the dimension panel for backward compatibility. So it was decided to change the design of the component to match custom padding when presets are provided (excluding the settings icon): https://github.com/WordPress/gutenberg/pull/56185#discussion_r1398989244

<img width="281" alt="Screenshot 2023-11-20 at 10 29 10" src="https://github.com/WordPress/gutenberg/assets/846565/3cb7d916-277e-44e9-a90d-eec785f88fc5">

## How?


Since this component isn't used in many places, the main goal was to match the design specs so we could move forward with updating to the 40px sizing. So there could be potential follow-ups to reduce some redundancy currently present, but I don't believe it is a priority due to its current usage. Therefore, the main changes are: 
- changing flex-direction to column
- adding a `RangeControl` component to each side
- adjusting where/when the icons are shown
- grouping the buttons together (reset and linked)

## Screenshots 

### AllInputControl

| Before  | After |
| ------------- | ------------- |
|  <img width="290" alt="Screenshot 2023-11-30 at 5 13 29 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/e14d89d3-20b1-456b-859a-765e846f4265"> |  <img width="269" alt="Screenshot 2023-11-30 at 5 30 42 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/b1b75e98-d013-4b86-90eb-c9aa4440650b"> |

### InputControls

| Before  | After |
| ------------- | ------------- |
|  <img width="270" alt="Screenshot 2023-11-30 at 5 13 36 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/dd6636ac-c110-419a-a012-ec0609375b3c"> | <img width="276" alt="Screenshot 2023-12-01 at 6 31 05 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/2b262120-56f5-4866-a16e-22e8e16f200e">  | 

### AxialInputControls

Will look similar but with two inputs when unlinked and the icons highlighting two sides (vertical and horizontal)

<img width="283" alt="Screenshot 2023-12-01 at 6 31 13 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/65614cda-4574-4e50-849b-336049213d39">

## Testing Instructions

Ensure tests pass and component behaves the same as before. 

For new changes: 

Ensure that RangeControl updates to the correct value when typing in the input.
